synapseclient/models/agent.py

```diff
@@ -0,0 +1,447 @@
+from dataclasses import dataclass, field
+from datetime import datetime
+from enum import Enum
+from typing import Dict, List, Optional
+
+from synapseclient import Synapse
+from synapseclient.api import (
+    get_agent,
+    get_response,
+    get_session,
+    get_trace,
+    register_agent,
+    send_prompt,
+    start_session,
+    update_session,
+)
+from synapseclient.core.async_utils import otel_trace_method
+
+
+class AgentType(Enum):
+    """
+    Enum representing the type of agent as defined in
+        <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/agent/AgentType.html>
+    'BASELINE' is a default agent provided by Synapse.
+    'CUSTOM' is a custom agent that has been registered by a user.
+    """
+
+    BASELINE = "BASELINE"
+    CUSTOM = "CUSTOM"
+
+
+class AgentSessionAccessLevel(Enum):
+    """
+    Enum representing the access level of the agent session as defined in
+        <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/agent/AgentAccessLevel.html>
+    """
+
+    PUBLICLY_ACCESSIBLE = "PUBLICLY_ACCESSIBLE"
+    READ_YOUR_PRIVATE_DATA = "READ_YOUR_PRIVATE_DATA"
+    WRITE_YOUR_PRIVATE_DATA = "WRITE_YOUR_PRIVATE_DATA"
+
+
+@dataclass
+class AgentPrompt:
+    """Represents a prompt, response, and metadata within an AgentSession.
+
+    Attributes:
+        id: The unique ID of the agent prompt.
+        prompt: The prompt to send to the agent.
+        response: The response from the agent.
+        trace: The trace of the agent session.
+    """
+
+    id: Optional[str] = None
+    """The unique ID of the agent prompt."""
+
+    prompt: Optional[str] = None
+    """The prompt sent to the agent."""
+
+    response: Optional[str] = None
+    """The response from the agent."""
+
+    trace: Optional[str] = None
+    """The trace or "though process" of the agent when responding to the prompt."""
+
+
+@dataclass
+class AgentSession:
+    """Represents a [Synapse Agent Session](https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/agent/AgentSession.html)
+
+    Attributes:
+        id: The unique ID of the agent session. Can only be used by the user that created it.
+        access_level: The access level of the agent session.
+            One of PUBLICLY_ACCESSIBLE, READ_YOUR_PRIVATE_DATA, or WRITE_YOUR_PRIVATE_DATA.
+        started_on: The date the agent session was started.
+        started_by: The ID of the user who started the agent session.
+        modified_on: The date the agent session was last modified.
+        agent_registration_id: The registration ID of the agent that will be used for this session.
+        etag: The etag of the agent session.
+    """
+
```

Comment:
Check out the tables refactor PR for examples of the syntax for an Examples section. I'd like for the docstring to have scripts (Copy/pastable) that folks could run and get started.